app/actions/editor.ts

```diff
@@ -28,13 +35,30 @@ export function changeAvro(value: string): ChangeAvroAction {
   };
 }
 
+export function changeAvroIsInError(
+  isInError: boolean
+): ChangeAvroIsInErrorAction {
+  return {
+    type: CHANGE_AVRO_IS_IN_ERROR,
+    value: isInError
+  };
+}
+
 export function changeAvroWithDispatch(
   value: string
 ): (dispatch: Dispatch) => void {
   return (dispatch: Dispatch) => {
     dispatch(changeAvro(value));
 
-    const jsonFromAvro = avro2json(JSON.parse(value));
-    dispatch(changeJson(JSON.stringify(jsonFromAvro, null, 4)));
+    try {
+      const avroAsObject = JSON.parse(value);
+      avro.Type.forSchema(avroAsObject);
+
+      const jsonFromAvro = avro2json(avroAsObject);
+      dispatch(changeJson(JSON.stringify(jsonFromAvro, null, 4)));
+      dispatch(changeAvroIsInError(false));
```

Comment:
The state mutation `isInError` to `false` should be done in the reducer :octocat: